libfwupdplugin/fu-common.c

```diff
@@ -3384,3 +3384,119 @@ fu_xmlb_builder_insert_kb (XbBuilderNode *bn, const gchar *key, gboolean value)
 {
 	xb_builder_node_insert_text (bn, key, value ? "true" : "false", NULL);
 }
+
+/**
+ * fu_common_get_firmware_search_path:
+ * @error: (nullable): optional return location for an error
+ *
+ * Reads the FU_PATH_KIND_FIRMWARE_SEARCH and
+ * returns its contents
+ *
+ * Returns: a pointer to a gchar array
+ *
+ * Since: 1.6.2
+ **/
+gchar *
+fu_common_get_firmware_search_path (GError **error)
+{
+	gsize sz = 0;
+	g_autofree gchar *sys_fw_search_path = NULL;
+	g_autofree gchar *contents = NULL;
+
+	sys_fw_search_path = fu_common_get_path (FU_PATH_KIND_FIRMWARE_SEARCH);
+	if (!g_file_get_contents(sys_fw_search_path, &contents, &sz, error))
+		return NULL;
+
+	/* remove newline character */
+	if (contents != NULL && sz > 0 && contents[sz - 1] == '\n')
+		contents[sz - 1] = 0;
+
+	g_debug ("read firmware search path (%" G_GSIZE_FORMAT "): %s", sz, contents);
+
+	return g_steal_pointer (&contents);
+}
+
+/**
+ * fu_common_set_firmware_search_path:
+ * @path: string to write
+ * @path_sz: number of bytes to write
+ * @error: (nullable): optional return location for an error
+ *
+ * Writes to the FU_PATH_KIND_FIRMWARE_SEARCH
+ *
+ * Returns: %TRUE if successful
+ *
+ * Since: 1.6.2
+ **/
+gboolean
+fu_common_set_firmware_search_path (const gchar *path, gsize path_sz, GError **error)
```

Comment:
Can we just document the "path is NUL-terminated, and then we can drop the path_sz argument. We can so the strlen in this function, and drop it from the callers. With that change I think the PR looks good to me and is probably ready to be merged. Good work!

Reply:
I still want to test this work myself, but I don't have much free time at the moment, so my test should definitely not block merging all this if it's already reviewed :D

Reply:
@hughsie added the comment and dropped the path_sz argument. Thank you for all of the valuable feedback!

@aleksander0m thanks! it seems there is a change request still pending
![image](https://user-images.githubusercontent.com/46465609/125522322-9f470369-0589-44b1-8023-adcdf86e2502.png)

Reply:
Just approved the changes according to what you did w.r.t. my last review

Reply:
thanks!